# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/nsCSPContext.h
# Commit: 0785d6bfebb3
# Full Hash: 0785d6bfebb3405227469a255389dac80b627f91
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2025-01-17 21:57:54
# Regressor Bug: 1901492
# File Overlap Count: 1
# Description:
#   Bug 1901492 - Introduce RequireTrustedTypesForDirectiveState enum for CSP. r=smaug
#   
#   Extend the hasPolicyWithRequireTrustedTypesForDirective boolean on
#   nsIContentSecurityPolicy to an enum in order to indicate whether there
#   is actually a require-trusted-types-for directive with "enforce"
# ==============================================================================

diff -r 23ca1773d73a -r 0785d6bfebb3 dom/security/nsCSPContext.h
--- a/dom/security/nsCSPContext.h	Fri Jan 17 13:35:39 2025 +0000
+++ b/dom/security/nsCSPContext.h	Fri Jan 17 13:35:39 2025 +0000
@@ -215,7 +215,8 @@
   mozilla::TimeStamp mSendReportLimitSpanStart;
   uint32_t mSendReportLimitCount = 1;
   bool mWarnedAboutTooManyReports = false;
-  bool mHasPolicyWithRequireTrustedTypesForDirective = false;
+  RequireTrustedTypesForDirectiveState mRequireTrustedTypesForDirectiveState =
+      RequireTrustedTypesForDirectiveState::NONE;
 };
 
 // Class that listens to violation report transmission and logs errors.